[Dexter Leng] iP

dexterleng opened the pull request:
# Pookie: Your Ultimate Task Management Tool 🚀  
Pookie is a **lightweight**, **text-based** task manager designed to free your mind from the burden of remembering tasks. As David Allen wisely said:  

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://gettingthingsdone.com))  

With Pookie, you can focus on what _truly matters_ while it handles the rest. Here's why you'll love it:  

* **Simple and intuitive**: Easy to learn and use.  
* **Blazing fast**: SUPER FAST performance for seamless task management.  
* **Free forever**: No hidden costs, no subscriptions.  

## Key Features  
- All of them :D

## Getting Started  
To start using Pookie, follow these steps:  

1. Download it from [here](#).  
2. Double-click the application to launch it.  
3. Add your tasks using the simple text interface.  
4. Let Pookie manage your tasks for you. 😉  

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

dexterleng committed:
Rename Duke to Pookie

dexterleng committed:
greet the user and exit

dexterleng committed:
Add echo functionality and exit command to chatbot

dexterleng committed:
Add task management with add and list commands

dexterleng committed:
Add support for marking tasks as done and undone

dexterleng committed:
reuse invalid task number error handling

dexterleng committed:
Implement task types: ToDos, Deadlines, and Events

dexterleng committed:
Add test

dexterleng committed:
Handle invalid usage for todo, deadline, and event

dexterleng committed:
Add test cases for invalid usages of todo, deadline, and event

dexterleng committed:
Use ArrayList to hold tasks

dexterleng committed:
Add delete command; update tests

dexterleng committed:
Persistence: Save and load tasks from disk

dexterleng committed:
Parse, save, and print dates using LocalDateTime

dexterleng committed:
Add Storage class

dexterleng committed:
Create TaskList class to manage tasks

dexterleng committed:
Storage: parameterise file

dexterleng committed:
Make Pookie do its work in an instance instead of static methods

dexterleng committed:
Move classes into separate files

dexterleng committed:
Delete unused variable

dexterleng committed:
Move command logic to separate Command classes

dexterleng committed:
Add a Ui class responsible for printing

dexterleng committed:
Move read line functionality from Pookie to Ui

dexterleng committed:
Merge branch 'add-gradle-support'

dexterleng committed:
Add tests for DeadlineCommand and EventCommand

dexterleng committed:
gradle: Update jar filename and main class

dexterleng committed:
Move files into packages (command and model)

dexterleng committed:
Add JavaDocs

dexterleng committed:
Style: Do not indent case branches and never import with *

dexterleng committed:
Merge branch 'branch-A-CodingStandard'

dexterleng committed:
Add find command and its tests

dexterleng committed:
Merge branch 'branch-Level-9'

dexterleng committed:
Fix merge issues

dexterleng committed:
FindCommandTest: rename test cases

dexterleng committed:
Add JavaFx GUI

dexterleng committed:
Delete text-ui-test

Jeremiah33-3 commented on src/main/java/pookie/Parser.java:
```diff
@@ -0,0 +1,50 @@
+package pookie;
```
I thought it might be neater if Parser can have a package of itself 😄

dexterleng replied:
I opted not to do so because Parser is a single file. If it somehow expands to multiple files then I will def do it.

Jeremiah33-3 commented on src/main/java/pookie/Pookie.java:
```diff
@@ -0,0 +1,88 @@
+    private Ui ui;
+    private Storage storage;
+    private TaskList tasks;
+    private boolean testMode;
```
perhaps it will be clearer if you can name the `boolean` variable to match what it is supposed to check? (e.g. isTestMode)

dexterleng replied:
addressed!

Jeremiah33-3 commented on src/main/java/pookie/Storage.java:
```diff
@@ -0,0 +1,119 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
```
well done on grouping your imports neatly and consistently for most of your classes! 👍

dexterleng replied:
thank u boss!

Jeremiah33-3 commented on src/main/java/pookie/Storage.java:
```diff
@@ -0,0 +1,119 @@
+                Task task = null;
+                switch (parts[0]) {
+                case "T":
+                    if (parts.length != 3) throw new CorruptFileException();
```
I thought it might be more consistent to our [coding standard guide](https://se-education.org/guides/conventions/java/intermediate.html#package-and-import-statements) to put conditional on a separate line!

dexterleng replied:
addressed!

Jeremiah33-3 reviewed (commented):
Some minor feedbacks on coding standard, overall looks good! Keep up the progress 💯

dexterleng committed:
Rename testMode to isTestMode

dexterleng committed:
Put conditionals on a separate line

dexterleng committed:
Add assertions to TaskList

dexterleng committed:
Move ui code into pookie/ui

dexterleng committed:
Merge pull request #1 from dexterleng/branch-A-Assertions

Add assertions

dexterleng committed:
Merge branch 'master' into branch-A-CodeQuality

dexterleng committed:
Merge pull request #2 from dexterleng/branch-A-CodeQuality

Move ui code into pookie/ui

dexterleng committed:
Rename duke to pookie

dexterleng committed:
Make the GUI actually work

dexterleng committed:
Create one letter aliases for todo, deadline, event, and find command

dexterleng committed:
Save messages when application quits through means other than /bye

dexterleng committed:
Correct indentation for task addition messages

The indentation for the task addition messages within EventCommand.java was inconsistent, leading to potential formatting issues or lack of readability in the UI. This commit fixes the alignment to maintain a consistent style when displaying messages to the user.

dexterleng committed:
Delete ConsoleUi

ConsoleUi has been replaced by the GUI and is no longer needed.

dexterleng committed:
Make UI constrainted to window size; Pin text field and button to the bottom

dexterleng committed:
Set the window title to 'Pookie'

dexterleng committed:
Add product screenshot

dexterleng committed:
Add README
